site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/repository/SiteStore.java

```diff
@@ -0,0 +1,65 @@
+package uk.ac.ox.ndph.mts.site_service.repository;
+
+import org.hl7.fhir.r4.model.Organization;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.ResearchStudy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.ac.ox.ndph.mts.site_service.converter.EntityConverter;
+import uk.ac.ox.ndph.mts.site_service.model.Site;
+
+/**
+ * Implement an EntityStore for Site.
+ */
+@Component
+public class SiteStore implements EntityStore<Site> {
+
+    private final FhirRepository repository;
+    private final EntityConverter<Site, org.hl7.fhir.r4.model.Organization> converter;
+    private final Logger logger = LoggerFactory.getLogger(SiteStore.class);
+
+    /**
+     *
+     * @param repository - The fhir repository
+     * @param converter - a model-entity to fhir-entity converter
+     */
+    @Autowired
+    public SiteStore(FhirRepository repository,
+                     EntityConverter<Site, org.hl7.fhir.r4.model.Organization> converter) {
+        this.repository = repository;
+        this.converter = converter;
+    }
+
+    /**
+     *
+     * @param entity - The Site entity
+     * @return String
+     */
+    @Override
+    public String saveEntity(Site entity) {
```

Comment:
would this be better accepting an Organization with the conversion being done before passing to a save method as this method name is incorrect - it states we are saving a Site when in fact we are saving an Organization.

Reply:
Taking forward as tech debt.